app/src/main/java/org/openobservatory/ooniprobe/model/database/Url.java

```diff
@@ -56,6 +75,43 @@ public static Url checkExistingUrl(String input, String categoryCode, String cou
 		return url;
 	}
 
+	public static List<String> saveOrUpdate(List<Url> urls) {
```

Comment:
This function is named `saveOrUpdate` _but_ it also returns a list of strings from the list of URLs. Therefore I suggest to either document this extra operation that we also perform (which is probably done here to avoid recomputing twice the same value) or to improve the function naming to clarify what it does. If I am given a function named `saveOrUpdate` without extra documentation, I'd be a bit surprised that it _also_ returns something because I'd expect it to just save or update.

If you choose to go down the commenting road, here's an example of what I mean (feel free to write a better Javadoc string if you find the one that I have written is not so good, I meant to write it as an example):

```suggestion
	// This function saves or updates input urls into the database and returns the
	// list of strings used internally to perform this operation, where each string
	// is the `.url` value of each URL. We return this list of strings because they are
	// actually useful to continue processing the input URLs.
	public static List<String> saveOrUpdate(List<Url> urls) {
```